diff --git a/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/eclipse/model/internal/EclipseDependenciesCreatorTest.groovy b/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/eclipse/model/internal/EclipseDependenciesCreatorTest.groovy
--- a/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/eclipse/model/internal/EclipseDependenciesCreatorTest.groovy
+++ b/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/eclipse/model/internal/EclipseDependenciesCreatorTest.groovy
@@ -23,7 +23,7 @@
 import org.gradle.util.TestUtil
 
 class EclipseDependenciesCreatorTest extends AbstractProjectBuilderSpec{
-    private final ProjectInternal project = TestUtil.createRootProject(temporaryFolder.testDirectory)
+    ProjectInternal project = TestUtil.createRootProject(temporaryFolder.testDirectory)
     private final ProjectInternal childProject = TestUtil.createChildProject(project, "child", new File("."))
     private final EclipseClasspath eclipseClasspath = new EclipseClasspath(project)
     private final dependenciesProvider = new EclipseDependenciesCreator(eclipseClasspath)
diff --git a/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/idea/model/internal/IdeaDependenciesProviderTest.groovy b/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/idea/model/internal/IdeaDependenciesProviderTest.groovy
--- a/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/idea/model/internal/IdeaDependenciesProviderTest.groovy
+++ b/subprojects/ide/src/test/groovy/org/gradle/plugins/ide/idea/model/internal/IdeaDependenciesProviderTest.groovy
@@ -31,7 +31,7 @@
 import org.gradle.util.TestUtil
 
 class IdeaDependenciesProviderTest extends AbstractProjectBuilderSpec {
-    private final ProjectInternal project = TestUtil.createRootProject(temporaryFolder.testDirectory)
+    ProjectInternal project = TestUtil.createRootProject(temporaryFolder.testDirectory)
     private final ProjectInternal childProject = TestUtil.createChildProject(project, "child", new File("."))
     def serviceRegistry = new DefaultServiceRegistry()
         .add(LocalComponentRegistry, Stub(LocalComponentRegistry))
diff --git a/subprojects/launcher/src/test/groovy/org/gradle/launcher/continuous/ContinuousIntegrationTestFixtureTest.groovy b/subprojects/launcher/src/test/groovy/org/gradle/launcher/continuous/ContinuousIntegrationTestFixtureTest.groovy
--- a/subprojects/launcher/src/test/groovy/org/gradle/launcher/continuous/ContinuousIntegrationTestFixtureTest.groovy
+++ b/subprojects/launcher/src/test/groovy/org/gradle/launcher/continuous/ContinuousIntegrationTestFixtureTest.groovy
@@ -26,7 +26,7 @@
 class ContinuousIntegrationTestFixtureTest extends Specification {
     def "successful build should be parsed"() {
         given:
-        def sampleTest = new AbstractContinuousIntegrationTest() {}
+        def sampleTest = new SampleTest()
         def gradleHandle = setupStubs(sampleTest)
         gradleHandle.getStandardOutput() >> '''
 Continuous build is an incubating feature.
@@ -44,7 +44,7 @@
 
     def "failure output should be parsed"() {
         given:
-        def sampleTest = new AbstractContinuousIntegrationTest() {}
+        def sampleTest = new SampleTest()
         def gradleHandle = setupStubs(sampleTest)
         gradleHandle.getStandardOutput() >> '''
 Continuous build is an incubating feature.
@@ -66,7 +66,7 @@
 
     def "output parsing with info logging after 'waiting for changes' line"() {
         given:
-        def sampleTest = new AbstractContinuousIntegrationTest() {}
+        def sampleTest = new SampleTest()
         def gradleHandle = setupStubs(sampleTest)
         gradleHandle.getStandardOutput() >> '''
 Continuous build is an incubating feature.
@@ -85,7 +85,7 @@
 
     def "output parsing of multiple continuous builds"() {
         given:
-        def sampleTest = new AbstractContinuousIntegrationTest() {}
+        def sampleTest = new SampleTest()
         def gradleHandle = setupStubs(sampleTest)
         gradleHandle.getStandardOutput() >> '''
 Continuous build is an incubating feature.
@@ -124,4 +124,5 @@
         gradleHandle.isRunning() >>>  [true, false]
         gradleHandle
     }
+    static class SampleTest extends AbstractContinuousIntegrationTest {}
 }
--- a/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransformsTest.groovy
+++ b/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransformsTest.groovy
@@ -179,8 +179,8 @@
         consumerSchema.withProducer(producerSchema) >> attributeMatcher
         attributeMatcher.matches(_, _) >> []
 
-        matchingCache.collectConsumerVariants(_, _, _) >> { AttributeContainerInternal from, AttributeContainerInternal to, ConsumerVariantMatchResult result ->
-                result.matched(to, Stub(ArtifactTransformer), 1)
+        matchingCache.collectConsumerVariants(_, _, _) >> { AttributeContainerInternal from, AttributeContainerInternal to, ConsumerVariantMatchResult r ->
+                r.matched(to, Stub(ArtifactTransformer), 1)
         }
 
         def selector = transforms.variantSelector(typeAttributes("dll"), true)
--- a/subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/state/InputPropertiesSerializerTest.groovy
+++ b/subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/state/InputPropertiesSerializerTest.groovy
@@ -33,7 +33,7 @@
     @Subject serializer = new InputPropertiesSerializer()
 
     def "serializes empty properties"() {
-        write [:]
+        write([:])
         expect:
         [:] == written
     }
