--- a/subprojects/logging/src/test/groovy/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandlerTest.groovy
+++ b/subprojects/logging/src/test/groovy/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandlerTest.groovy
@@ -69,7 +69,7 @@
         event.logLevel == LogLevel.WARN
     }
 
-    def "no warnings should be displayed in #mode"() {
+    def "no warnings should be displayed in #type"() {
         when:
         handler.init(locationReporter, type)
         handler.deprecatedFeatureUsed(deprecatedFeatureUsage('feature1'))
--- a/subprojects/base-services/src/test/groovy/org/gradle/internal/time/TimeFormattingTest.groovy
+++ b/subprojects/base-services/src/test/groovy/org/gradle/internal/time/TimeFormattingTest.groovy
@@ -54,7 +54,7 @@
     }
 
     @Unroll
-    def "shows #output when elapsed time is greater or equals than #lowerBoundInclusive but lower than #upperBoundExlusive"() {
+    def "shows #output when elapsed time is greater or equals than #lowerBoundInclusive but lower than #upperBoundExclusive"() {
         when:
         def result = TimeFormatting.formatDurationTerse(input)
 
