--- a/subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceFileWatcherBacking.java
+++ b/subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceFileWatcherBacking.java
@@ -117,7 +117,7 @@
                 public void onFailure(Throwable t) {
                     running.set(false);
                 }
-            });
+            }, com.google.common.util.concurrent.MoreExecutors.directExecutor());
             return fileWatcher;
         } else {
             throw new IllegalStateException("file watcher is started");
--- a/subprojects/base-services-groovy/src/main/java/org/gradle/groovy/scripts/internal/AstUtils.java
+++ b/subprojects/base-services-groovy/src/main/java/org/gradle/groovy/scripts/internal/AstUtils.java
@@ -204,6 +204,7 @@
             public boolean apply(ScriptBlock input) {
                 return names.contains(input.getName());
             }
+            public boolean test(ScriptBlock input) { return apply(input); }
         });
     }
 
