TODO: Fix the relevant Guix package instead
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -59,7 +59,7 @@
 libraries.jetty = "org.mortbay.jetty:jetty:6.1.26"
 libraries.commons_codec = "commons-codec:commons-codec:1.6"
 libraries.jcifs = "jcifs:jcifs:1.3.17"
-libraries.commons_httpclient = "org.apache.httpcomponents:httpclient:${versions.httpclient}"
+libraries.commons_httpclient = ["org.apache.httpcomponents:httpclient:${versions.httpclient}", "org.apache.httpcomponents:httpcore:${versions.httpclient}"]
 
 
 libraries += [
