--- a/settings.gradle
+++ b/settings.gradle
@@ -14,9 +14,6 @@
  * limitations under the License.
  */
 
-apply {
-    from("gradle/remoteHttpCacheSettings.gradle")
-}
 
 include("distributions")
 include("baseServices")
--- a/buildSrc/settings.gradle.kts
+++ b/buildSrc/settings.gradle.kts
@@ -14,6 +14,3 @@
  * limitations under the License.
  */
 
-apply {
-    from("../gradle/remoteHttpCacheSettings.gradle")
-}
