--- a/subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceFileWatcherBacking.java
+++ b/subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceFileWatcherBacking.java
@@ -117,7 +117,7 @@
                 public void onFailure(Throwable t) {
                     running.set(false);
                 }
-            });
+            }, com.google.common.util.concurrent.MoreExecutors.directExecutor());
             return fileWatcher;
         } else {
             throw new IllegalStateException("file watcher is started");
Avoid dependency on errorprone annotations
--- a/gradle/strictCompile.gradle
+++ b/gradle/strictCompile.gradle
@@ -17,7 +17,7 @@
 allprojects {
     ext.strictCompile = {
         tasks.withType(AbstractCompile) {
-            options.compilerArgs << "-Werror" << "-Xlint:all" << "-Xlint:-options" << "-Xlint:-serial"
+            options.compilerArgs << "-Werror" << "-Xlint:all" << "-Xlint:-classfile" << "-Xlint:-options" << "-Xlint:-serial"
         }
     }
     ext.strictCompileIgnoreDeprecations = {
--- a/subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaInstallationProbe.java
+++ b/subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaInstallationProbe.java
@@ -18,7 +18,7 @@
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
-import com.google.common.io.Files;
+import java.nio.file.Files;
 import org.apache.commons.io.FileUtils;
 import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
@@ -141,7 +141,12 @@
     private EnumMap<SysProp, String> getMetadataInternal(File jdkPath) {
         JavaExecAction exec = factory.newJavaExecAction();
         exec.executable(javaExe(jdkPath, "java"));
-        File workingDir = Files.createTempDir();
+        File workingDir;
+        try {
+            workingDir = Files.createTempDirectory(null).toFile();
+        } catch (IOException e) {
+            throw new GradleException("Unable to create temp directory", e);
+        }
         exec.setWorkingDir(workingDir);
         exec.setClasspath(new SimpleFileCollection(workingDir));
         try {
