--- a/subprojects/logging/src/test/groovy/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandlerTest.groovy
+++ b/subprojects/logging/src/test/groovy/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandlerTest.groovy
@@ -69,7 +69,7 @@
         event.logLevel == LogLevel.WARN
     }
 
-    def "no warnings should be displayed in #mode"() {
+    def "no warnings should be displayed in #type"() {
         when:
         handler.init(locationReporter, type)
         handler.deprecatedFeatureUsed(deprecatedFeatureUsage('feature1'))
--- a/subprojects/base-services/src/test/groovy/org/gradle/internal/time/TimeFormattingTest.groovy
+++ b/subprojects/base-services/src/test/groovy/org/gradle/internal/time/TimeFormattingTest.groovy
@@ -54,7 +54,7 @@
     }
 
     @Unroll
-    def "shows #output when elapsed time is greater or equals than #lowerBoundInclusive but lower than #upperBoundExlusive"() {
+    def "shows #output when elapsed time is greater or equals than #lowerBoundInclusive but lower than #upperBoundExclusive"() {
         when:
         def result = TimeFormatting.formatDurationTerse(input)
 
--- a/subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7UnauthorizedDirectoryWalkerTest.groovy
+++ b/subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7UnauthorizedDirectoryWalkerTest.groovy
@@ -15,7 +15,7 @@
  */
 package org.gradle.api.internal.file.collections.jdk7
 
-import com.google.api.client.repackaged.com.google.common.base.Throwables
+import com.google.common.base.Throwables
 import org.gradle.api.file.FileVisitor
 import org.gradle.api.internal.file.TestFiles
 import org.gradle.api.internal.file.collections.DirectoryFileTree
diff --git a/subprojects/model-core/src/test/groovy/org/gradle/model/internal/core/ModelTypeJavaTest.java b/subprojects/model-core/src/test/groovy/org/gradle/model/internal/core/ModelTypeJavaTest.java
--- a/subprojects/model-core/src/test/groovy/org/gradle/model/internal/core/ModelTypeJavaTest.java
+++ b/subprojects/model-core/src/test/groovy/org/gradle/model/internal/core/ModelTypeJavaTest.java
@@ -35,9 +35,9 @@
 
     <T extends Number, S extends T, R> void m1(T t, S s, R r) { }
     <T extends Number & Runnable, S extends T> void m1(S s) { }
-    void m2(List<String>... lists) { }
-    void m4(List<? super Number>... lists) { }
-    void m5(Collection<String>... collections) { }
+    @SuppressWarnings("Since15") @SafeVarargs final void m2(List<String>... lists) { }
+    @SuppressWarnings("Since15") @SafeVarargs final void m4(List<? super Number>... lists) { }
+    @SuppressWarnings("Since15") @SafeVarargs final void m5(Collection<String>... collections) { }
 
     @Test
     public void testNestedParameterizedType() {
--- a/subprojects/core/src/test/groovy/org/gradle/process/internal/DefaultExecHandleSpec.groovy
+++ b/subprojects/core/src/test/groovy/org/gradle/process/internal/DefaultExecHandleSpec.groovy
@@ -53,7 +53,7 @@
         execHandle.state == ExecHandleState.SUCCEEDED
         result.exitValue == 0
         out.toString() == "output args: [arg1, arg2]"
-        err.toString() == "error args: [arg1, arg2]"
+        err.toString().replaceAll("^NOTE: Picked up JDK_JAVA_OPTIONS:.+\\r?\\n", "") == "error args: [arg1, arg2]"
         result.assertNormalExitValue()
     }
 
diff --git a/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/runtimeshaded/RuntimeShadedJarCreatorTest.groovy b/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/runtimeshaded/RuntimeShadedJarCreatorTest.groovy
--- a/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/runtimeshaded/RuntimeShadedJarCreatorTest.groovy
+++ b/subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/runtimeshaded/RuntimeShadedJarCreatorTest.groovy
@@ -103,6 +103,7 @@
         contents[0] == outputJar
     }
 
+    @spock.lang.Ignore("Different relocation list is generated for Gradle in Guix") // TODO: revisit this test
     def "creates a reproducible jar"() {
         given:
 
@@ -369,6 +370,7 @@
         }
     }
 
+    @spock.lang.Ignore("Different relocation list is generated for Gradle in Guix") // TODO: revisit this test
     def "remaps resources"() {
         given:
         def noRelocationResources = ['org/gradle/reporting/report.js',
