--- a/spock-core/src/main/java/spock/lang/Specification.java
+++ b/spock-core/src/main/java/spock/lang/Specification.java
@@ -43,7 +43,7 @@
    * Example: <tt>[foo, _] &lt;&lt; loadDataFromDb()</tt>
    * </ul>
    */
-  public static final Object _ = Wildcard.INSTANCE;
+  public static Object get_() { return Wildcard.INSTANCE; }
 
   /**
    * Specifies that the preceding <tt>when</tt> block should throw an exception.
