--- a/src/test/groovy/net/rubygrapefruit/platform/FileSystemsTest.groovy
+++ b/src/test/groovy/net/rubygrapefruit/platform/FileSystemsTest.groovy
@@ -29,6 +29,7 @@
         Native.get(FileSystems.class) == fileSystems
     }
 
+    @spock.lang.Ignore("/etc/mtab is not available inside build in Guix")
     def "can query filesystem details"() {
         expect:
         fileSystems.fileSystems.collect() { it.mountPoint }.containsAll(File.listRoots())
--- a/src/test/groovy/net/rubygrapefruit/platform/AbstractFilesTest.groovy
+++ b/src/test/groovy/net/rubygrapefruit/platform/AbstractFilesTest.groovy
@@ -28,9 +28,6 @@
     }
 
     long toJavaFileTime(long time) {
-        if (Platform.current().isLinux()) {
-            return (time / 1000).longValue() * 1000 // round to nearest second
-        }
         return time
     }
 }
--- a/src/test/groovy/net/rubygrapefruit/platform/FilesTest.groovy
+++ b/src/test/groovy/net/rubygrapefruit/platform/FilesTest.groovy
@@ -122,6 +122,7 @@
         file << File.listRoots()
     }
 
+    @spock.lang.Ignore("/etc/mtab is not available inside build in Guix")
     def "can stat the file system roots reported by OS"() {
         expect:
         def stat = files.stat(fileSystem.mountPoint)
